pkg/sypgp/sypgp.go

```diff
@@ -105,61 +113,84 @@ func PublicPath() string {
 	return filepath.Join(DirPath(), "pgp-public")
 }
 
-// PathsCheck creates the sypgp home folder, secret and public keyring files
-func PathsCheck() error {
-	// create the sypgp base directory
-	if err := os.MkdirAll(DirPath(), 0700); err != nil {
+func ensureDirPrivate(dn string) error {
+	mode := os.FileMode(0700)
+
+	oldumask := syscall.Umask(0077)
+
+	err := os.MkdirAll(dn, mode)
+
+	// restore umask...
+	syscall.Umask(oldumask)
+
+	// ... and check if there was an error
+
+	if err != nil {
 		return err
 	}
```

Comment:
This err check should probably go right under line 121 (`err := os.MkdirAll(dn, mode)`). *Move `syscall.Umask(oldumask)` right under the err check.*

Reply:
no, I want to restore the old umask before potentially bailing out of the function. on line 129.

Reply:
Okay, that makes sense :+1: